include/ProjectVersion.h

```diff
@@ -28,29 +29,102 @@
+		m_build(s.section( '.', 2 ).section( '-', 1 )),
+		m_compareType(c)
 	{
 	}
 
-	static int compare( const ProjectVersion & _v1,
-						const ProjectVersion & _v2 );
+	static int compare(const ProjectVersion & v1, const ProjectVersion & v2);
+
+	const int getMajor() const { return m_major; }
+	const int getMinor() const { return m_minor; }
+	const int getRelease() const { return m_release; }
+	const QString getBuild() const { return m_build; }
+	const CompareType getCompareType() const { return m_compareType; }
+	
 
+private:
+	const int m_major;
+	const int m_minor;
+	const int m_release;
+	const QString m_build;
+	const CompareType m_compareType;
 } ;
 
 
 
 
-inline bool operator<( const ProjectVersion & _v1, const char * _str )
+inline int compStr(const ProjectVersion & v1, const char * v2)
+{
+	return ProjectVersion::compare(v1, ProjectVersion(v2));
+}
+
+
+
+inline int compStr(const ProjectVersion & v1, const QString & v2)
 {
-	return ProjectVersion::compare( _v1, ProjectVersion( _str ) ) < 0;
+	return ProjectVersion::compare(v1, ProjectVersion(v2));
 }
 
 
 
+/*
+ * ProjectVersion v. char[]
+ */
+inline bool operator<(const ProjectVersion & v1, const char * v2) { return compStr(v1, v2) < 0; }
+inline bool operator>(const ProjectVersion & v1, const char * v2) { return compStr(v1, v2) > 0; }
+inline bool operator<=(const ProjectVersion & v1, const char * v2) { return compStr(v1, v2) <= 0; }
+inline bool operator>=(const ProjectVersion & v1, const char * v2) { return compStr(v1, v2) >= 0; }
+inline bool operator==(const ProjectVersion & v1, const char * v2) { return compStr(v1, v2) == 0; }
+inline bool operator!=(const ProjectVersion & v1, const char * v2) { return compStr(v1, v2) != 0; }
+
+inline bool operator<(const char * v1, const ProjectVersion & v2) { return 0 < compStr(v2, v1); }
+inline bool operator>(const char * v1, const ProjectVersion & v2) { return 0 > compStr(v2, v1); }
+inline bool operator<=(const char * v1, const ProjectVersion & v2) { return 0 <= compStr(v2, v1); }
+inline bool operator>=(const char * v1, const ProjectVersion & v2) { return 0 >= compStr(v2, v1); }
+inline bool operator==(const char * v1, const ProjectVersion & v2) { return 0 == compStr(v2, v1); }
+inline bool operator!=(const char * v1, const ProjectVersion & v2) { return 0 != compStr(v2, v1); }
+
+/*
+ * ProjectVersion v. QString
+ */
+inline bool operator<(const ProjectVersion & v1, const QString & v2) { return compStr(v1, v2) < 0; }
+inline bool operator>(const ProjectVersion & v1, const QString & v2) { return compStr(v1, v2) > 0; }
+inline bool operator<=(const ProjectVersion & v1, const QString & v2) { return compStr(v1, v2) <= 0; }
+inline bool operator>=(const ProjectVersion & v1, const QString & v2) { return compStr(v1, v2) >= 0; }
+inline bool operator==(const ProjectVersion & v1, const QString & v2) { return compStr(v1, v2) == 0; }
+inline bool operator!=(const ProjectVersion & v1, const QString & v2) { return compStr(v1, v2) != 0; }
```

Comment:
> The compiler should be able to automatically convert the types by using the ProjectVersion(const QString&) constructor.

That would be nice.  I'm pessimistic, but I'll test that a bit as it would really cleanup these redundant functions. :+1: